CalibTracker/SiPhase2TrackerESProducers/plugins/SiPhase2BadStripConfigurableFakeESSource.cc

```diff
@@ -0,0 +1,255 @@
+
+ Description: "fake" SiStripBadStrip ESProducer - configurable list of bad modules
+
+ Implementation:
+     Adapted to Phase-2 from CalibTracker/SiStripESProducers/plugins/fake/SiStripBadModuleConfigurableFakeESSource.cc
+*/
+
+// system include files
+#include <memory>
+
+// user include files
+#include "CondFormats/DataRecord/interface/SiPhase2OuterTrackerCondDataRecords.h"
+#include "CondFormats/SiStripObjects/interface/SiStripBadStrip.h"
+#include "DataFormats/Phase2TrackerDigi/interface/Phase2TrackerDigi.h"
+#include "DataFormats/SiStripDetId/interface/SiStripDetId.h"
+#include "DataFormats/SiStripDetId/interface/StripSubdetector.h"
+#include "DataFormats/TrackerCommon/interface/TrackerTopology.h"
+#include "FWCore/Framework/interface/ESProducer.h"
+#include "FWCore/Framework/interface/EventSetupRecordIntervalFinder.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "FWCore/Utilities/interface/RandomNumberGenerator.h"
+#include "Geometry/CommonDetUnit/interface/PixelGeomDetUnit.h"
+#include "Geometry/CommonTopologies/interface/PixelTopology.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "Geometry/Records/interface/TrackerTopologyRcd.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+
+// neede for the random number generation
+#include "CLHEP/Random/RandFlat.h"
+#include "CLHEP/Random/JamesRandom.h"
+
+class SiPhase2BadStripConfigurableFakeESSource : public edm::ESProducer, public edm::EventSetupRecordIntervalFinder {
+public:
+  SiPhase2BadStripConfigurableFakeESSource(const edm::ParameterSet&);
+  ~SiPhase2BadStripConfigurableFakeESSource() override = default;
+
+  void setIntervalFor(const edm::eventsetup::EventSetupRecordKey&,
+                      const edm::IOVSyncValue& iov,
+                      edm::ValidityInterval& iValidity) override;
+
+  typedef std::unique_ptr<SiStripBadStrip> ReturnType;
+  ReturnType produce(const SiPhase2OuterTrackerBadStripRcd&);
+
+  static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+private:
+  std::map<unsigned short, unsigned short> clusterizeBadChannels(
+      const std::vector<Phase2TrackerDigi::PackedDigiType>& maskedChannels);
+
+  // configurables
+  bool m_printDebug;
+  float m_badComponentsFraction;
+
+  // random engine
+  std::unique_ptr<CLHEP::HepRandomEngine> m_engine;
+
+  // es tokens
+  edm::ESGetToken<TrackerTopology, TrackerTopologyRcd> m_trackTopoToken;
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> m_geomToken;
```

Comment:
Hi Marco, I thought that trailing underscore, i.e. like `printDebug_`, `geomToken_` is the preferred style, based on 2.9 in https://cms-sw.github.io/cms_coding_rules.html

Reply:
fixed in the last push.